packages/node_modules/@node-red/editor-client/src/js/ui/deploy.js

```diff
@@ -653,9 +657,13 @@ RED.deploy = (function() {
                     ensureUnlocked(ws.z)
                     ws.changed = false;
                     delete ws.added
+                    if (flowsToLock.has(ws)) {
+                        ws.locked = true;
```

Comment:
```suggestion
                        ws.locked = true;
                        flowsToLock.delete(ws);
```

To keep things tidy, remove from flowsToLock so we don't lock it again later. It's pretty harmless, but keeps `flowsToLock` as an accurate set of things to be locked later.